@xen-orchestra/web/src/components/vdis/VmVdisTable.vue

```diff
@@ -0,0 +1,158 @@
+              <div v-if="column.id === 'name'" class="name">
+                <VtsIcon size="medium" name="fa:hard-drive" />
+                <div v-tooltip class="text-ellipsis">
+                  {{ column.value }}
+                </div>
+              </div>
+              <div v-else-if="column.id === 'used-space'">
+                <VtsSizeProgressCell :current="column.value.used" :total="column.value.total" />
+              </div>
+              <div v-else-if="column.id === 'size'">
+                <div class="size">
+                  {{ column.value }}
+                </div>
+              </div>
+              <div v-else v-tooltip class="text-ellipsis">
+                {{ column.value }}
+              </div>
+            </td>
+          </tr>
+        </template>
+      </VtsDataTable>
+      <VtsStateHero v-if="searchQuery && filteredVdis.length === 0" format="table" type="no-result" size="small">
+        {{ t('no-result') }}
+      </VtsStateHero>
+      <UiTopBottomTable :selected-items="0" :total-items="0">
+        <UiTablePagination v-bind="paginationBindings" />
+      </UiTopBottomTable>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { IconName } from '@core/icons'
+import VtsDataTable from '@core/components/data-table/VtsDataTable.vue'
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import VtsSizeProgressCell from '@core/components/size-progress-cell/VtsSizeProgressCell.vue'
+import VtsStateHero from '@core/components/state-hero/VtsStateHero.vue'
+import UiQuerySearchBar from '@core/components/ui/query-search-bar/UiQuerySearchBar.vue'
+import UiTablePagination from '@core/components/ui/table-pagination/UiTablePagination.vue'
+import UiTitle from '@core/components/ui/title/UiTitle.vue'
+import UiTopBottomTable from '@core/components/ui/top-bottom-table/UiTopBottomTable.vue'
+import { usePagination } from '@core/composables/pagination.composable.ts'
+import { useRouteQuery } from '@core/composables/route-query.composable.ts'
+import { useTable } from '@core/composables/table.composable.ts'
+import { vTooltip } from '@core/directives/tooltip.directive.ts'
+import { formatSize } from '@core/utils/size.util.ts'
+import type { XoVdi } from '@vates/types'
+import { computed, ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+const { vdis } = defineProps<{
+  vdis: XoVdi[]
+  hasError: boolean
+  isReady: boolean
+}>()
+
+const { t } = useI18n()
+
+const selectedVdiId = useRouteQuery('id')
+
+const searchQuery = ref('')
+
+const filteredVdis = computed(() => {
+  const searchTerm = searchQuery.value.trim().toLocaleLowerCase()
+
+  if (!searchTerm) {
+    return vdis
+  }
+
+  return vdis.filter(vdi => Object.values(vdi).some(value => String(value).toLocaleLowerCase().includes(searchTerm)))
+})
+
+const getFormat = (format: string | undefined) => (format ? format.toUpperCase() : t('vhd'))
```

Comment:
Just in case we need to extract this in a helper, the name will be more explicit:
```suggestion
const getVdiFormat = (format: string | undefined) => (format ? format.toUpperCase() : t('vhd'))
```